lib/evmone/advanced_instructions.cpp

```diff
@@ -161,66 +170,12 @@ const Instruction* opx_beginblock(const Instruction* instr, AdvancedExecutionSta
 constexpr std::array<instruction_exec_fn, 256> instruction_implementations = []() noexcept {
     std::array<instruction_exec_fn, 256> table{};
 
-    table[OP_STOP] = op<stop>;
-    table[OP_ADD] = op<add>;
-    table[OP_MUL] = op<mul>;
-    table[OP_SUB] = op<sub>;
-    table[OP_DIV] = op<div>;
-    table[OP_SDIV] = op<sdiv>;
-    table[OP_MOD] = op<mod>;
-    table[OP_SMOD] = op<smod>;
-    table[OP_ADDMOD] = op<addmod>;
-    table[OP_MULMOD] = op<mulmod>;
-    table[OP_EXP] = op<exp>;
-    table[OP_SIGNEXTEND] = op<signextend>;
-    table[OP_LT] = op<lt>;
-    table[OP_GT] = op<gt>;
-    table[OP_SLT] = op<slt>;
-    table[OP_SGT] = op<sgt>;
-    table[OP_EQ] = op<eq>;
-    table[OP_ISZERO] = op<iszero>;
-    table[OP_AND] = op<and_>;
-    table[OP_OR] = op<or_>;
-    table[OP_XOR] = op<xor_>;
-    table[OP_NOT] = op<not_>;
-    table[OP_BYTE] = op<byte>;
-    table[OP_SHL] = op<shl>;
-    table[OP_SHR] = op<shr>;
-    table[OP_SAR] = op<sar>;
-
-    table[OP_KECCAK256] = op<keccak256>;
-
-    table[OP_ADDRESS] = op<address>;
-    table[OP_BALANCE] = op<balance>;
-    table[OP_ORIGIN] = op<origin>;
-    table[OP_CALLER] = op<caller>;
-    table[OP_CALLVALUE] = op<callvalue>;
-    table[OP_CALLDATALOAD] = op<calldataload>;
-    table[OP_CALLDATASIZE] = op<calldatasize>;
-    table[OP_CALLDATACOPY] = op<calldatacopy>;
-    table[OP_CODESIZE] = op<codesize>;
-    table[OP_CODECOPY] = op<codecopy>;
-    table[OP_GASPRICE] = op<gasprice>;
-    table[OP_EXTCODESIZE] = op<extcodesize>;
-    table[OP_EXTCODECOPY] = op<extcodecopy>;
-    table[OP_RETURNDATASIZE] = op<returndatasize>;
-    table[OP_RETURNDATACOPY] = op<returndatacopy>;
-    table[OP_EXTCODEHASH] = op<extcodehash>;
-    table[OP_BLOCKHASH] = op<blockhash>;
-    table[OP_COINBASE] = op<coinbase>;
-    table[OP_TIMESTAMP] = op<timestamp>;
-    table[OP_NUMBER] = op<number>;
-    table[OP_DIFFICULTY] = op<difficulty>;
-    table[OP_GASLIMIT] = op<gaslimit>;
-    table[OP_CHAINID] = op<chainid>;
-    table[OP_SELFBALANCE] = op<selfbalance>;
-    table[OP_BASEFEE] = op<basefee>;
-
-    table[OP_POP] = op<pop>;
-    table[OP_MLOAD] = op<mload>;
-    table[OP_MSTORE] = op<mstore>;
-    table[OP_MSTORE8] = op<mstore8>;
-    table[OP_SLOAD] = op<sload>;
+    // Init table with wrapped generic implementations.
+#define X(OPCODE, IDENTIFIER) table[OPCODE] = op<IDENTIFIER>;
+    MAP_OPCODE_TO_IDENTIFIER
+#undef X
+
+    // Overwrite with Advanced-specific implementations.
     table[OP_SSTORE] = op_sstore;
```

Comment:
Why is sstore overridden for the advanced interpreter but not sload?

Reply:
All instructions that need access to gas counter are replaced. This information is not directly available in the Advanced so we need to produce it for them. In particular `SSTORE` needs it if `gas < stipend`.

Reply:
No. I.e. many instruction are doing additional gas checks, but they don't  need to know exact value.